terraform/aws/best-practice/aws-elasticache-automatic-backup-not-enabled.tf

```diff
@@ -0,0 +1,41 @@
+# pass
+
+resource "aws_elasticache_cluster" "enabled" {
+  cluster_id           = "cluster"
+  engine               = "redis"
+  node_type            = "cache.m5.large"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.x"
+
+  snapshot_retention_limit = 5
+}
+
+# failure
+# ruleid: aws-elasticache-automatic-backup-not-enabled
+resource "aws_elasticache_cluster" "default" {
+  cluster_id           = "cluster"
+  engine               = "redis"
+  node_type            = "cache.m5.large"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.x"
+}
+# ruleid: aws-elasticache-automatic-backup-not-enabled
+resource "aws_elasticache_cluster" "disabled" {
+  cluster_id           = "cluster"
+  engine               = "redis"
+  node_type            = "cache.m5.large"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.x"
+
+  snapshot_retention_limit = 0
```

Comment:
The associated rule will not flag this one since the rule only checks for the presence of `snapshot_retention_limit` set to anything. You can filter this out with another `pattern-not-inside` clause checking for `0`.